feat: make analysis buttons open workflow landing pages (#137)

hunterckx opened the pull request:
- ~~No error handling for the API request~~
- ~~Only regulation is set up~~

mvdbeek commented on app/utils/galaxy-api.ts:
```diff
@@ -0,0 +1,44 @@
+import ky from "ky";
+
+interface WorkflowLandingsBody {
+  request_state: { reference_genome: string };
```
This is going to vary across workflows, I suppose
```suggestion
  request_state: { [key: string]: string }:
```
would be the right interface ?

mvdbeek commented on app/utils/galaxy-api.ts:
```diff
@@ -0,0 +1,44 @@
+      reference_genome: referenceGenome,
+    },
+    workflow_id: workflowId,
+    workflow_target_type: "trs_url",
```
```suggestion
    workflow_target_type: "trs_url",
    public: true,
```

mvdbeek commented on app/utils/galaxy-api.ts:
```diff
@@ -0,0 +1,44 @@
+    },
+  });
+  const id = (await res.json()).uuid;
+  return WORKFLOW_LANDING_URL_PREFIX + encodeURIComponent(id);
```
```suggestion
  return `${WORKFLOW_LANDING_URL_PREFIX}${id}?public=true;
```

mvdbeek reviewed (commented):
Pretty cool thank you!

hunterckx commented:
@mvdbeek Thanks for the feedback! I've:
- Updated the types to make more room for alternative keys (though in a different way than suggested -- I wanted to make it explicit what the keys can be)
- Added `public: true` to the API request body
- Added `?public=true` to the constructed landing page URL (but left the URL encoding in -- although it's not necessary in theory, I'm inclined to be cautious and default to encoding things when constructing URLs)

NoopDog commented:
@hunterckx, can you rebase this, please? Thanks!

mvdbeek commented on app/apis/catalog/brc-analytics-catalog/common/entities.ts:
```diff
@@ -25,3 +34,8 @@ export interface EntitiesResponsePagination {
 }
+
+export enum WORKFLOW_ID {
+  REGULATION = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/ChIPseq_PE/main",
```
```suggestion
  REGULATION = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/chipseq-pe/main/versions/v0.12",
```

mvdbeek commented on app/apis/catalog/brc-analytics-catalog/common/entities.ts:
```diff
@@ -25,3 +34,8 @@ export interface EntitiesResponsePagination {
+
+export enum WORKFLOW_ID {
+  REGULATION = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/ChIPseq_PE/main",
+  TRANSCRIPTOMICS = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/rnaseq-pe/main",
```
```suggestion
  TRANSCRIPTOMICS = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/rnaseq-pe/main/versions/v0.9",
```

mvdbeek reviewed (commented):
We're expecting the full TRS url here, which is the TRS id and the version.

mvdbeek commented:
For right now this is working well. I have a few things to build on here, could we get this merged ?
